File: P_4/Napakalaki_Java_P4/src/napakalaki/Prize.java
// MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

package napakalaki;

/*
* Clase Prize del paquete napakalaki
*/

public class Prize{
    
    // Declaramos los atributos PRIVADOS a utilizar
    private final int level;
    private final int treasures;

    // Constructor
    public Prize(int treasure, int level){
        
        this.treasures = treasure;
        this.level = level;
    }

    // Consultores básicos
    // Devuelve los tesoros que se pueden ganar con el buen rollo
    public int getTreasures(){
        return treasures;
    }

    // Devuelve los niveles que pueden ganarse con el buen rollo
    public int getLevel(){
        return level;
    }

    // Método toString para pintar una secuencia
    public String toString(){
        return "Level -> " + level + ", Treasures -> " + treasures;
    }
}

File: P_4/Napakalaki_Ruby_P4/lib/prize.rb
#encoding: utf-8
# MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

module NapakalakiRuby

class Prize
    
    attr_reader :treasures
    attr_reader :level
    
    # Constructor
    def initialize(treasures = 0, level = 0)
        # Atributos
        @treasures = treasures
        @level = level
    end

    # Metodo to_string para pintar en pantalla lo deseado
    def to_s
        "Tesoros ganados: #{@treasures.to_s}
        Niveles ganados: #{@level.to_s}"
    end
    
    def mayor_a_uno
        return @level > 1
    end
    
end

end
